File: 프로그래머스/3/12927. 야근 지수/야근 지수.java
import java.util.*;

class Solution {
    public long solution(int n, int[] works) {
        long answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int work : works) pq.add(work);
        for(int i = 0; i < n; i++){
            int work = pq.poll();
            if(work == 0) return 0;
            pq.add(work - 1);
        }
        while(!pq.isEmpty()){
            answer += Math.pow(pq.poll(), 2);
        }
        return answer;
    }
}

File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq

def solution(n, works):
    # 최대 힙을 구현하기 위해 음수로 변환
    pq = [-work for work in works]
    heapq.heapify(pq)
    
    for _ in range(n):
        if not pq or pq[0] == 0:
            return 0  # 남은 작업이 없으면 바로 0 반환
        max_work = -heapq.heappop(pq)
        max_work -= 1
        heapq.heappush(pq, -max_work)
    
    # 남은 작업의 제곱합 계산
    return sum((-work) ** 2 for work in pq)
